src/hotspot/share/gc/serial/defNewGeneration.hpp

```diff
@@ -138,6 +138,10 @@ class DefNewGeneration: public Generation {
   ContiguousSpace* _eden_space;
   ContiguousSpace* _from_space;
   ContiguousSpace* _to_space;
+  // Saved mark word
+  HeapWord* _eden_saved_mark_word;
+  HeapWord* _from_saved_mark_word;
```

Comment:
Are they really needed? I'd expect `saved_mark_word` to be useful only for copy-destination, i.e. to-space and old-gen.

Reply:
Before, I tried to keep the previous meaning and didn't delete the existing code.

The `save_mark_word` of eden-space and from-space only influence the assertment. I removed them just now.